### ISSUE TITLE
Изменение структуры наборов геоданных

### ISSUE THREAD

**Issue by ANAT01** [nextgis/quickmapservices#25]
Сейчас для описания набора нужно создать папку в которой разместить файлы metadata.ini и файл иконки. В метадате указать имя файла иконки и id набора.

Предлагаю делать описание геоданных одним файлом без создания отдельных папок. Его имя будет служить идентификатором (id), картинку прописывать в нем же icon = "..." в виде base64 кодировке, например.

Также вместо отдельного описания группы (group =) набора предлагаю использовать префикс в имени.

Например, файл rosreestr_basemap.ini
значит группа у него "rosreestr"
id у него "rosreestr_basemap"
в файле прописать icon = /9j/4AAQSkZJRgABAQAAAQABAAD/4QAqRXhpZgAASUkqAAgAAAABADEBAgAHAAAAGgAAAAAAAABHb29nbGUAAP/bAIQAAwICAwICAwMDAwQDAwQFCAUFBAQFCgcHBggMCgwPCwoMCw0ODhANDhEODQsQFhARFA4VFRUNDxcYFhQYEhQUDQEDBAQGBQYKBgYKDg0MDQ8NDxAMDBAMDQ0PDBQMFA0NDQ4PDA8UDQ4QDQ0PDA0NDBANDQ0PDBQMDRANDBQNDg0N/8AAEQgAMAAwAwERAAIRAQMRAf/EABwAAAEFAQEBAAAAAAAAAAAAAAcBAgQICQYDAP/EADUQAAEDAwIEAgcHBQAAAAAAAAECAwQABREGIRIiMUEHkRMyM1FxgdEIFlJTYWPBFBclNGL/xAAbAQEAAwEBAQEAAAAAAAAAAAAAAQYHAgMFBP/EAC4RAAICAQEEBwkBAAAAAAAAAAABAgMEEQUhMUESEzJRUnHRFCJCkaGxweHxI//aAAwDAQACEQMRAD8A0TAzXkepV77XH2s3vCZz7paRW0vVbrQckzVpDiLchQ5eU7KdUOYA5AGCQeICrTsfZHtX+tvYXBeJ+n8KntrbPsq6qrtvn4f2Ug05qq9ax8VtL3C/XabeZq7xEJfnPqdUMvo2HETgfoMAVe7aK6seca4pLoy4LTkUCnJsuyYSsk2+kuPmaLrefiXaW9HfcjuB5fM2opPrGscRtYQNGaxXeVf0M/AmpTlDgGA8B127KFNAdSoYB+FQBQcAk9BvtQGSms9A+JmutX3rUUzQWrlSbpMdlq47HKJSFqJSn2fRIwkDsAK1/Gvxaao1qyG5JdqPqY3l4mXfbKx1T3tvgz00F4Qa+ha80zIf0Hqllhq6xFuOuWSUlKEh5BKiS3gADck0ys3HdM0rI9mXxLu8xibPyY3QbrkkpLk+80Hk2S6GdKUm1zlJU6shQjLIIyf0rIUzZBI9uvECbGlt2qfxsOJWMRl74O46dxkVOoDC5uDiuQKjc0ACYoywjJPT310NSdbR/koe59u33/6FCNSsF/Ur7wXUcav9t7v+4axW9vrZ7/il9yrTfvPzZJ0dYJGrdV2mzsca1S5CUKwTyt5ytXySFH5V6YtUr7oVx13tfLn9NTqqLnJRRf1Z2ONq2MtAqO9SAGxY74ZQDHe6flq+ldAm25h4XGJlh0D06Ny2fxChGhXz+32ptVanuTdrsU6SFzHsPFkttAekO5cVhI86yB4WRfdJVwk/ee/TRce97it9TOcn0U+LLLeC3gox4aR13Ce43Nv8hHAt1v2bCO6EZ3OTjKts4Gw73zZWyY4a6c983z5Jdy/L5n2MfH6re+IUFHIPwqwn7RyO9ABKNcpqmUkzpXT89f1rohk+zO3C432BGRNlEKeSVj0yjyA5V39wNCQvqVvXIGE5oBD0PwoByFDPrDzoAIWmDLuSENw4rshWAMoTyj4q6D510Am6Q0qjTrS5EhaHJ7qcKKTyoT+EfyfpvDYOgK059YedQBONPvHnQHxUnhPMOnvoD//Z

**Comment by yellow-sky** [comment]
К сожалению, перекодирование картинки в base64 доступно не всем. И это не позволит использовать (в будущем) одни иконки для различных сервисов (при одинаковых картинках).

Структура метаданных будет еще пересматриваться.
Возможно предложенные изменения будут приняты.

**Comment by ANAT01** [comment]
Я согласен что это может вогнать в ступор, но есть куча онлайн перекодировщиков типа http://www.base64-image.de/step-1.php так что не думаю что это большая проблема.
Да, может быть это чуть сложнее чем просто копировать файлы.

**Comment by ANAT01** [comment]
Возраждаю предложение, а именно: предлагаю хранить все пользовательсвик сервисы в одном json файле.

Это шаг в сторону отказа унификации в работе с пользовательскими и серверными списками сервисов идея описана [здесь](http://gis-lab.info/forum/viewtopic.php?f=65&t=18464&p=149400#p149400)  и [здесь](http://gis-lab.info/forum/viewtopic.php?f=65&t=18464&p=149400#p149399)
